### ISSUE TITLE
Missing Japanese Language Support

### ISSUE THREAD

**Issue by StefanGreve** [migueldeicaza/gui.cs#1661]
**Describe the bug**
The TUI views are completely broken if the text for labels etc. are in Japanese. It's still possible to use the controls (such as buttons), so this "only" affects the rendering in the terminal. To be clear: In my development environment, Japanese characters are still displayed correctly in the console. Its the border drawing and view alignment that's broken (under these specific circumstances).

**To Reproduce**

To display Japanese characters in the terminal on a machine using a English culture, I added these lines to my PowerShell profile:

```powershell
chcp 932 | Out-Null
$PSDefaultParameterValues['*:Encoding'] = 'utf8'
```

I use Windows Terminal and Microsoft's [Cascadia Mono font](https://github.com/microsoft/cascadia-code) which supports rendering Japanese characters.

**Expected behavior**
I didn't really expect this to work seeing how tricky Japanese font faces can be in a terminal-based environment, so this would be a nice-to-have feature and a heads-up for everyone looking to write a TUI with Japanese localization support.

**Screenshots**
Login view with slight modifications from the reactive sample code and Japanese text. 

<img width="574" alt="image" src="https://user-images.githubusercontent.com/32100623/162828650-2fd57e05-0310-4042-83c1-ef80a73a93c5.png">

**Desktop (please complete the following information):**
 - OS: Windows 
 - Version: 10.0.19044

**Additional context**
As mentioned previously, the code to reproduce this error can be obtained by replacing the label content with Japanese text using the the `ReactiveUI` demo project. This problem potentially also affects all other major Asian Languages (CJK), would be interesting to know if something similar also happens with other difficult languages such as Arabic or Hebrew.

**Comment by StefanGreve** [comment]
Wow thanks a bunch, guess I grew a little to dependent on Windows Terminal that it didn't cross my mind to try a different terminal emulator. Hope your PR goes through!

**Comment by StefanGreve** [comment]
This might also be interesting to other Japanese developers:

The output looks good in Developer PowerShell for VS 2022 if you add

```csharp
public static void ChangeCulture()
{
    Thread.CurrentThread.CurrentUICulture = new CultureInfo(Configuration.Language);
    Thread.CurrentThread.CurrentCulture = CultureInfo.CreateSpecificCulture(Configuration.Language);

    // You only need this line in .NET Core; install System.Text.CodePages from the Nuget Manager
    Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);

    Console.OutputEncoding = Configuration.Language switch
    {
        Language.Japanese => Encoding.GetEncoding(932),
        _ => Encoding.Default
    };
}
 ```

For Windows Terminal, I also added

```csharp
public static void Main(string[] args)
{
    if (args.Length > 0 && args.Contains("-usc"))
    {
        Application.UseSystemConsole = true;
    }

    // etc.
```

I got this idea from the `demo.cs` file [here](https://github.com/migueldeicaza/gui.cs/blob/2a29ca520db01841130bc9fced4c0701da763597/Example/demo.cs#L568). Now we only have to wait for your PR and this should solve the issue with Windows Terminal as well.

**Comment by tig** [comment]
Fixed with #1664
